tests/test_ec2/test_security_groups.py

```diff
@@ -563,6 +563,48 @@ def test_authorize_all_protocols_with_no_port_specification():
     permission.shouldnt.have.key("ToPort")
 
 
+@mock_ec2
+def test_create_and_describe_security_grp_rule():
+    ec2 = boto3.resource("ec2", "us-east-1")
+    client = boto3.client("ec2", "us-east-1")
+    vpc = ec2.create_vpc(CidrBlock="10.0.0.0/16")
+    sg_name = str(uuid4())
+    sg = ec2.create_security_group(
+        Description="Test SG", GroupName=sg_name, VpcId=vpc.id
+    )
+
+    # Ingress rule
+    ip_permissions = [
+        {
+            "IpProtocol": "tcp",
+            "FromPort": 27017,
+            "ToPort": 27017,
+            "IpRanges": [{"CidrIp": "1.2.3.4/32"}],
+        }
+    ]
+    sgr = sg.authorize_ingress(IpPermissions=ip_permissions)
+    # Describing the ingress rule
+    sgr_id = sgr['SecurityGroupRules'][0]['SecurityGroupRuleId']
+    response = client.describe_security_group_rules(
+        Filters=[
+            {
+                'Name': 'ip-permission-id',
+                'Values': [sgr_id]
+            }
+        ]
+    )
+    ingress_rule = response["SecurityGroupRules"]
```

Comment:
the `describe_security_group_rules` API call returns a list of all the rules that match the filter criteria, not just the one that you are interested in.  So the variable `ingress_rule` contains 4 results. I can't figure out the reason why bcz i'm not creating any additional rules. But it also contains the one in which we are interested so I've iterated through the list of rules and check if the security group rule id matches the id we are expecting. Once we find the expected rule, Then assertions are performed on it.

Let me know if this approach dosen't sounds good to you. And feel free to suggest a better one if you have in mind